### ISSUE TITLE
Build fails without pulseaudio

### ISSUE THREAD

**Issue by jigmpatel** [greshake/i3status-rust#752]
I am trying to build with --no-default-features as I don't use pulseaudio. However the build fails because unresolved import of libpulse_binding.

Please find below the output on the error:

error[E0432]: unresolved import `libpulse_binding`
 --> src/main.rs:3:5
  |
3 | use libpulse_binding as pulse;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^ no `libpulse_binding` external crate

warning: unused import: `lazy_static::lazy_static`
  --> src/blocks/sound.rs:30:5
   |
30 | use lazy_static::lazy_static;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error: aborting due to previous error; 1 warning emitted

**Comment by ammgws** [comment]
Looks like it was accidentally removed in 1be527ca65c08c07adb4094fabda0bf02402f1e7

Can you check with #753?

**Comment by jigmpatel** [comment]
Thanlks. I am able to build with #[cfg(feature = "pulseaudio")].

**Comment by jigmpatel** [comment]
The build fails with the latest pull I tried with the following error (initially I managed to build without pulling from the origin and manually adding #[[cfg(default)] for pulse lib import).

error[E0412]: cannot find type `SourceInfo` in this scope
   --> src/blocks/sound.rs:229:15
    |
229 | impl TryFrom<&SourceInfo<'_>> for PulseAudioVolInfo {
    |     -         ^^^^^^^^^^ not found in this scope
    |     |
    |     help: you might be missing a type parameter: `<SourceInfo>`

error[E0412]: cannot find type `PulseAudioVolInfo` in this scope
   --> src/blocks/sound.rs:229:35
    |
229 | impl TryFrom<&SourceInfo<'_>> for PulseAudioVolInfo {
    |                                   ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `SourceInfo` in this scope
   --> src/blocks/sound.rs:232:31
    |
232 |     fn try_from(source_info: &SourceInfo) -> std::result::Result<Self, Self::Error> {
    |                               ^^^^^^^^^^ not found in this scope

error[E0422]: cannot find struct, variant or union type `PulseAudioVolInfo` in this scope
   --> src/blocks/sound.rs:235:30
    |
235 |             Some(name) => Ok(PulseAudioVolInfo {
    |                              ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `SinkInfo` in this scope
   --> src/blocks/sound.rs:244:15
    |
244 | impl TryFrom<&SinkInfo<'_>> for PulseAudioVolInfo {
    |     -         ^^^^^^^^ not found in this scope
    |     |
    |     help: you might be missing a type parameter: `<SinkInfo>`

error[E0412]: cannot find type `PulseAudioVolInfo` in this scope
   --> src/blocks/sound.rs:244:33
    |
244 | impl TryFrom<&SinkInfo<'_>> for PulseAudioVolInfo {
    |                                 ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `SinkInfo` in this scope
   --> src/blocks/sound.rs:247:29
    |
247 |     fn try_from(sink_info: &SinkInfo) -> std::result::Result<Self, Self::Error> {
    |                             ^^^^^^^^ not found in this scope

error[E0422]: cannot find struct, variant or union type `PulseAudioVolInfo` in this scope
   --> src/blocks/sound.rs:250:30
    |
250 |             Some(name) => Ok(PulseAudioVolInfo {
    |                              ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `PULSEAUDIO_DEFAULT_SINK` in this scope
   --> src/blocks/sound.rs:698:27
    |
698 |             Self::Sink => PULSEAUDIO_DEFAULT_SINK.lock().unwrap().to_string(),
    |                           ^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `PULSEAUDIO_DEFAULT_SOURCE` in this scope
   --> src/blocks/sound.rs:699:29
    |
699 |             Self::Source => PULSEAUDIO_DEFAULT_SOURCE.lock().unwrap().to_string(),
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

warning: unused import: `TryInto`
  --> src/blocks/sound.rs:25:29
   |
25 | use std::convert::{TryFrom, TryInto};
   |                             ^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error: aborting due to 10 previous errors; 1 warning emitted

Some errors have detailed explanations: E0412, E0422, E0425.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `i3status-rs

**Comment by ammgws** [comment]
OK I can reproduce. No idea why it didn't come up when testing that PR

**Comment by ammgws** [comment]
Problem was some recently added code that wasn't guarded behind the pulseaudio feature flag.

#755 fixes it for good this time

**Comment by jigmpatel** [comment]
Thanks. I am able to build it now.

**Comment by NOTMY** [comment]
I can not build i3status-rust without pulse since commit: 4df33cc2483d63e62d2ac4bd790a26dfe128b781

```
$ cargo build --release --no-default-features
...
   Compiling i3status-rs v0.14.1 (/usr/local/src/i3status-rust)
error[E0425]: cannot find function `min` in this scope
   --> src/blocks/sound.rs:129:13
    |
129 |             min(new_vol, vol_cap)
    |             ^^^ not found in this scope
    |
help: consider importing one of these items
    |
24  | use core::cmp::min;
    |
24  | use std::cmp::min;
    |

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
error: could not compile `i3status-rs`.
```

**Comment by ammgws** [comment]
@NOTMY Fixed by #849
